### PR TITLE
test: end point to get intents across all zones for a delegator address

### PR DESCRIPTION
**Pull request by muku314115**
## 1. Summary
Fixes # (issue)

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

File: x/interchainstaking/keeper/grpc_query_test.go [hunk 1]
```diff
@@ -666,7 +794,7 @@ func (suite *KeeperTestSuite) TestKeeper_UserWithdrawalRecords() {
 				)
 			},
 			&types.QueryUserWithdrawalRecordsRequest{
-				UserAddress: testAddress,
+				UserAddress: "quick16pxh2v4hr28h2gkntgfk8qgh47pfmjfhzgeure",
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/grpc_query_test.go:797]: should this be hardcoded?

File: proto/quicksilver/interchainstaking/v1/query.proto [hunk 2]
```diff
@@ -143,6 +151,14 @@ message QueryDelegatorIntentResponse {
   DelegatorIntent intent = 1;
 }
 
+message QueryDelegatorIntentsRequest {
+  string delegator_address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+}
+
+message QueryDelegatorIntentsResponse {
+  repeated DelegatorIntent intents = 1;
```
Comment (joe-bowman) [inline proto/quicksilver/interchainstaking/v1/query.proto:159]: Does the DelegatorIntent contain the chain Id?

Comment (aljo242) [reply proto/quicksilver/interchainstaking/v1/query.proto:159]: It does not.

We should use something like: 

```proto
message DelegatorIntentsResponse {
     string chainID = 1;
     DelegatorIntent intent = 2;
}

message QueryDelegatorIntentsResponse {
  repeated DelegatorIntentsResponse intents = 1;
```

Comment (muku314115) [reply proto/quicksilver/interchainstaking/v1/query.proto]: added a chain id to the response as well with each intent

**Comment by muku314115** [comment]
> @muku314115 is the target branch correct?

Built this on top of the Withdrawal records PR, so set that as the target branch

https://github.com/ingenuity-build/quicksilver/pull/465 this would come as the first one

**Pull request closed without merging**